### PR TITLE
Improve Wifi configuration UI

### PR DESCRIPTION
**Pull request by MindFreeze**
## Proposed change

- Disable IP configuration until wifi network is selected
- Deduplicate wifi networks by name and show only the one with the strongest signal
- Reload IP configuration after save to show current IP
- Add clear button that resets the form to defaults and removes the wifi network. Essentially allowing to forget a network

image

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by matthiasdebaat** [comment]
CleanShot 2024-10-22 at 09 58 05@2x

A couple of suggestions:
- If there is no WiFi network connected, drop the IPv4 and IPv6 collapsibles.
- Move the "Clear" button to the W-Fi collapsible for better context.
- Change the clear button label to "Forget network"
- Change the "Scan for access points" label to "Search networks" as this is more common terminology.
- Move the W-Fi collapsible to the top, as the IP addresses are based on this selection.

**Comment by MindFreeze** [comment]
image

**Comment by MindFreeze** [comment]
I tried "reset to defaults" but I didn't like it. "Reset configuration" is better. I'll use that

**Review by piitaya — DISMISSED** [review]

File: src/panels/config/network/supervisor-network.ts [hunk 1]
```diff
@@ -566,6 +613,10 @@ export class HassioNetwork extends LitElement {
     }
     this._curTabIndex = ev.detail.index;
     this._interface = { ...this._interfaces[ev.detail.index] };
+    // mock
+    if (this._interface?.wifi) {
+      // this._interface!.wifi!.ssid = undefined;
+    }
```
Comment (piitaya) [inline src/panels/config/network/supervisor-network.ts:619]: Should be removed, right?

**Review by piitaya — APPROVED** [review]

**Pull request merged**